### PR TITLE
test: fixing lds flake

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low (test only)
Testing: yes: 1k runs now pass
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by phlax** [comment]
cc @soulxu

**Comment by phlax** [comment]
@alyssawilk seems like this has upset ci

**Comment by soulxu** [comment]
emm... I can't reproduce the original failure. But see other flaky. Let me try this fix whether works in my local

File: test/integration/listener_lds_integration_test.cc [hunk 1]
```diff
@@ -567,6 +567,8 @@ TEST_P(ListenerIntegrationTest, MultipleLdsUpdatesSharingListenSocketFactory) {
   test_server_->waitForCounterGe("listener_manager.listener_create_success", 1);
   // Make a connection to the listener from version 1.
   codec_client_ = makeHttpConnection(lookupPort(listener_name_));
+  // Ensure Envoy has accepted the connection before starting reloads.
+  test_server_->waitForCounterGe("listener.127.0.0.1_0.downstream_cx_total", 1);
```
Comment (soulxu) [inline test/integration/listener_lds_integration_test.cc:571]: This failed on IPv6 test, for ipv6 the stat name should be "listener.[__1]_0"

**Review by phlax — APPROVED** [review]
lgtm, thanks @alyssawilk

**Pull request merged**